Mathlib/Geometry/Manifold/IntegralCurve.lean

```diff
@@ -0,0 +1,288 @@
+/-
+Copyright (c) 2023 Winston Yin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Winston Yin
+-/
+import Mathlib.Analysis.ODE.PicardLindelof
+import Mathlib.Geometry.Manifold.InteriorBoundary
+import Mathlib.Geometry.Manifold.MFDeriv
+
+/-!
+# Integral curves of vector fields on a manifold
+
+For any continuously differentiable vector field on a manifold `M` and any chosen non-boundary point
+`x₀ : M`, there exists an integral curve `γ : ℝ → M` such that `γ t₀ = x₀` and the tangent vector of
+`γ` at `t` coincides with the vector field at `γ t` for all `t` within an open interval around `t₀`.
+
+As a corollary, such an integral curve exists for any starting point `x₀` if `M` is a manifold
+without boundary.
+
+## Main definition
+
+- **IsIntegralCurve γ v t₀ x₀**: If `v : M → TM` is a vector field on `M` and `x : M`,
+`IsIntegralCurveAt γ v t₀ x₀` means `γ : ℝ → M` is a differentiable integral curve of `v` with
+`γ x₀ = t₀`.
+
+## Tags
+
+integral curve, vector field, local existence
+-/
+
+open scoped Manifold
+open Set
+
+section
+
+variable
+  {𝕜 : Type*} [NontriviallyNormedField 𝕜]
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+  {H : Type*} [TopologicalSpace H] {I : ModelWithCorners 𝕜 E H}
+  {M : Type*} [TopologicalSpace M] [ChartedSpace H M] [SmoothManifoldWithCorners I M]
+  {E' : Type*} [NormedAddCommGroup E'] [NormedSpace 𝕜 E']
+  {H' : Type*} [TopologicalSpace H'] {I' : ModelWithCorners 𝕜 E' H'}
+  {M' : Type*} [TopologicalSpace M'] [ChartedSpace H' M'] [SmoothManifoldWithCorners I' M']
+
+lemma ModelWithCorners.isOpen_extend_target [I.Boundaryless] {e : LocalHomeomorph M H} :
```

Comment:
Looking at e.g. `LocalHomeomorph.isOpen_extend_source`, this should also be in the LocalHomeomorph namespace.